src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
```

Comment:
The task specifies not to use utility classes like `Objects`. Please replace `Objects.equals()` with a custom equality check.